utils/test/test.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+source scripts/test/config
+
+is_in_array()
+{
+    needle="$1"
+    haystack="$2"
+
+    found=0
+    for item in ${haystack[@]}; do
+        if test "$needle" = "$item"; then
+            found=1
+            break
+        fi
+    done
+
+    return "$found"
+}
+
+out_dir="scripts/test/$(date +"%Y-%m-%d-%H:%M:%S")"
```

Comment:
This looks to granular, you will very likely have a different directory for every test that is done during the testing phase (e.g. one directory for `qemu-x86_64-einitrd`, one for `fc-x86_64-einitrd`, etc.
I would stop at the hour mark.

Reply:
Won't work. You actually make multiple runs during an hour, while testing. And yes, the goal is for every test to have its own directory, you want to capture them all, right?